src/chains.rs

```diff
@@ -413,12 +413,8 @@ impl Chain {
             _ => expr.clone(),
         }
     }
-}
-
-impl Rewrite for Chain {
-    fn rewrite(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
-        debug!("rewrite chain {:?} {:?}", self, shape);
 
+    fn format(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
```

Comment:
I split this out into a separate function to more easily handle certain long-line scenarios when the `indent_style` is `Visual` and `allow_chain_call_overflow` is `true`, detailed below in the `rewrite` function